### ISSUE TITLE
HoneyClient package not installing properly

### ISSUE THREAD

**Issue by dkindlund** [dkindlund/honeyclient#44]
While trying to install a honeyclient, a user came across the following installation problem, and couldn't run the honeyclient as a result.

Here is the output after running Makefile.pl:

{{{
/usr/bin/perl.exe "-Iinc" Makefile.PL --config= --installdeps=Data::Compare,0,Data::Diff,0,Data::Structure::Util,0,DateTime::HiRes,0,Filesys::CygwinPaths,0,HTML::HeadParser,0,HTML::LinkExtor,0,HTTP::Request::Common,0,HoneyClient::Util::Config,0,HoneyClient::Util::SOAP,0,LWP::UserAgent,0,Log::Log4perl,0,Parse::Yapp::Driver,0,Search::Binary,0,Term::ProgressBar,0,Test::Exception,0,URI,0,URI::URL,0,Win32::Job,0
**\* Installing dependencies...
**\* Installing Data::Compare...
CPAN: Storable loaded ok
Going to read /home/hctest/.cpan/Metadata
  Database was generated on Thu, 28 Dec 2006 23:25:23 GMT
Running install for module Data::Compare
Running make for D/DC/DCANTRELL/Data-Compare-0.14.tar.gz
CPAN: Digest::MD5 loaded ok
CPAN: Compress::Zlib loaded ok
Checksum for /home/hctest/.cpan/sources/authors/id/D/DC/DCANTRELL/Data-Compare-0.14.tar.gz ok
Scanning cache /home/hctest/.cpan/build for sizes
Deleting from cache: /home/hctest/.cpan/build/libwin32-0.26 (11.0>10.0 MB)
Data-Compare-0.14/
Data-Compare-0.14/t/
Data-Compare-0.14/t/oo.t
Data-Compare-0.14/t/opts-ignore-hash-keys.t
Data-Compare-0.14/t/deep-recursion.t
Data-Compare-0.14/t/deep-objects.t
Data-Compare-0.14/t/plugins.t
Data-Compare-0.14/t/scalar-properties.t
Data-Compare-0.14/t/taint.t
Data-Compare-0.14/t/compare.t
Data-Compare-0.14/lib/
Data-Compare-0.14/lib/Data/
Data-Compare-0.14/lib/Data/Compare/
Data-Compare-0.14/lib/Data/Compare/Plugins/
Data-Compare-0.14/lib/Data/Compare/Plugins/Scalar/
Data-Compare-0.14/lib/Data/Compare/Plugins/Scalar/Properties.pm
Data-Compare-0.14/lib/Data/Compare/Plugins.pod
Data-Compare-0.14/lib/Data/Compare.pm
Data-Compare-0.14/MANIFEST
Data-Compare-0.14/TODO
Data-Compare-0.14/CHANGES
Data-Compare-0.14/MAINTAINERS-NOTE
Data-Compare-0.14/Makefile.PL
Data-Compare-0.14/README
Data-Compare-0.14/META.yml
Removing previously used /home/hctest/.cpan/build/Data-Compare-0.14

  CPAN.pm: Going to build D/DC/DCANTRELL/Data-Compare-0.14.tar.gz

Checking if your kit is complete...
Looks good
Writing Makefile for Data::Compare
---- Unsatisfied dependencies detected during [D/DC/DCANTRELL/Data-Compare-0.14.tar.gz] -----
    File::Find::Rule
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* Data::Compare installation failed.
**\* Installing Data::Structure::Util...
Running install for module Data::Structure::Util
Running make for F/FO/FOTANGO/Data-Structure-Util-0.12.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/F/FO/FOTANGO/Data-Structure-Util-0.12.tar.gz ok
Data-Structure-Util-0.12/
Data-Structure-Util-0.12/lib/
Data-Structure-Util-0.12/lib/Data/
Data-Structure-Util-0.12/lib/Data/Structure/
Data-Structure-Util-0.12/lib/Data/Structure/Util.xs
Data-Structure-Util-0.12/lib/Data/Structure/Util.pm
Data-Structure-Util-0.12/t/
Data-Structure-Util-0.12/t/06signature.t
Data-Structure-Util-0.12/t/01compile.t
Data-Structure-Util-0.12/t/05refs.t
Data-Structure-Util-0.12/t/04utf8.t
Data-Structure-Util-0.12/t/02circular.t
Data-Structure-Util-0.12/t/00pod.t
Data-Structure-Util-0.12/t/02circular_off.t
Data-Structure-Util-0.12/t/03bless.t
Data-Structure-Util-0.12/MANIFEST
Data-Structure-Util-0.12/CHANGES
Data-Structure-Util-0.12/META.yml
Data-Structure-Util-0.12/NINJA
Data-Structure-Util-0.12/bin/
Data-Structure-Util-0.12/bin/packages.pl
Data-Structure-Util-0.12/README
Data-Structure-Util-0.12/Makefile.PL
Data-Structure-Util-0.12/Build.PL
Removing previously used /home/hctest/.cpan/build/Data-Structure-Util-0.12

  CPAN.pm: Going to build F/FO/FOTANGO/Data-Structure-Util-0.12.tar.gz
# running Build.PL

/usr/bin/perl.exe Build.PL
Checking whether your kit is complete...
Looks good

Checking prerequisites...
Creating new 'Build' script for 'Data-Structure-Util' version '0.12'
---- Unsatisfied dependencies detected during [F/FO/FOTANGO/Data-Structure-Util-0.12.tar.gz] -----
    Clone
    Test::Pod
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* Data::Structure::Util installation failed.
**\* Installing DateTime::HiRes...
Running install for module DateTime::HiRes
Running make for J/JH/JHOBLITT/DateTime-HiRes-0.01.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/J/JH/JHOBLITT/DateTime-HiRes-0.01.tar.gz ok
DateTime-HiRes-0.01/
DateTime-HiRes-0.01/t/
DateTime-HiRes-0.01/t/01_load.t
DateTime-HiRes-0.01/t/02_now.t
DateTime-HiRes-0.01/lib/
DateTime-HiRes-0.01/lib/DateTime/
DateTime-HiRes-0.01/lib/DateTime/HiRes.pm
DateTime-HiRes-0.01/lib/DateTime/HiRes.pod
DateTime-HiRes-0.01/META.yml
DateTime-HiRes-0.01/Todo
DateTime-HiRes-0.01/Changes
DateTime-HiRes-0.01/MANIFEST
DateTime-HiRes-0.01/Build.PL
DateTime-HiRes-0.01/LICENSE
DateTime-HiRes-0.01/Makefile.PL
DateTime-HiRes-0.01/README
Removing previously used /home/hctest/.cpan/build/DateTime-HiRes-0.01

  CPAN.pm: Going to build J/JH/JHOBLITT/DateTime-HiRes-0.01.tar.gz
# running Build.PL

/usr/bin/perl.exe Build.PL
Checking whether your kit is complete...
Looks good

Checking prerequisites...
Creating new 'Build' script for 'DateTime-HiRes' version '0.01'
---- Unsatisfied dependencies detected during [J/JH/JHOBLITT/DateTime-HiRes-0.01.tar.gz] -----
    DateTime
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* DateTime::HiRes installation failed.
**\* Installing HTML::HeadParser...
Running install for module HTML::HeadParser
Running make for G/GA/GAAS/HTML-Parser-3.55.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/G/GA/GAAS/HTML-Parser-3.55.tar.gz ok
HTML-Parser-3.55/
HTML-Parser-3.55/t/
HTML-Parser-3.55/t/xml-mode.t
HTML-Parser-3.55/t/textarea.t
HTML-Parser-3.55/t/filter-methods.t
HTML-Parser-3.55/t/crashme.t
HTML-Parser-3.55/t/handler-eof.t
HTML-Parser-3.55/t/unicode-bom.t
HTML-Parser-3.55/t/argspec-bad.t
HTML-Parser-3.55/t/dtext.t
HTML-Parser-3.55/t/callback.t
HTML-Parser-3.55/t/entities.t
HTML-Parser-3.55/t/argspec.t
HTML-Parser-3.55/t/comment.t
HTML-Parser-3.55/t/declaration.t
HTML-Parser-3.55/t/offset.t
HTML-Parser-3.55/t/handler.t
HTML-Parser-3.55/t/plaintext.t
HTML-Parser-3.55/t/case-sensitive.t
HTML-Parser-3.55/t/cases.t
HTML-Parser-3.55/t/marked-sect.t
HTML-Parser-3.55/t/document.t
HTML-Parser-3.55/t/filter.t
HTML-Parser-3.55/t/tokeparser.t
HTML-Parser-3.55/t/linkextor-rel.t
HTML-Parser-3.55/t/entities2.t
HTML-Parser-3.55/t/skipped-text.t
HTML-Parser-3.55/t/linkextor-base.t
HTML-Parser-3.55/t/default.t
HTML-Parser-3.55/t/headparser.t
HTML-Parser-3.55/t/uentities.t
HTML-Parser-3.55/t/headparser-http.t
HTML-Parser-3.55/t/script.t
HTML-Parser-3.55/t/stack-realloc.t
HTML-Parser-3.55/t/pod.t
HTML-Parser-3.55/t/magic.t
HTML-Parser-3.55/t/options.t
HTML-Parser-3.55/t/attr-encoded.t
HTML-Parser-3.55/t/unicode.t
HTML-Parser-3.55/t/argspec2.t
HTML-Parser-3.55/t/ignore.t
HTML-Parser-3.55/t/largetags.t
HTML-Parser-3.55/t/unbroken-text.t
HTML-Parser-3.55/t/msie-compat.t
HTML-Parser-3.55/t/api_version.t
HTML-Parser-3.55/t/parsefile.t
HTML-Parser-3.55/t/parser.t
HTML-Parser-3.55/t/process.t
HTML-Parser-3.55/t/pullparser.t
HTML-Parser-3.55/eg/
HTML-Parser-3.55/eg/hlc
HTML-Parser-3.55/eg/hanchors
HTML-Parser-3.55/eg/htextsub
HTML-Parser-3.55/eg/htitle
HTML-Parser-3.55/eg/hstrip
HTML-Parser-3.55/eg/hrefsub
HTML-Parser-3.55/eg/hform
HTML-Parser-3.55/eg/hdump
HTML-Parser-3.55/eg/htext
HTML-Parser-3.55/lib/
HTML-Parser-3.55/lib/HTML/
HTML-Parser-3.55/lib/HTML/Entities.pm
HTML-Parser-3.55/lib/HTML/PullParser.pm
HTML-Parser-3.55/lib/HTML/Filter.pm
HTML-Parser-3.55/lib/HTML/TokeParser.pm
HTML-Parser-3.55/lib/HTML/LinkExtor.pm
HTML-Parser-3.55/lib/HTML/HeadParser.pm
HTML-Parser-3.55/util.c
HTML-Parser-3.55/Makefile.PL
HTML-Parser-3.55/hints/
HTML-Parser-3.55/hints/solaris.pl
HTML-Parser-3.55/Parser.pm
HTML-Parser-3.55/TODO
HTML-Parser-3.55/MANIFEST
HTML-Parser-3.55/tokenpos.h
HTML-Parser-3.55/Changes
HTML-Parser-3.55/mkhctype
HTML-Parser-3.55/Parser.xs
HTML-Parser-3.55/mkpfunc
HTML-Parser-3.55/hparser.c
HTML-Parser-3.55/README
HTML-Parser-3.55/hparser.h
HTML-Parser-3.55/typemap
Removing previously used /home/hctest/.cpan/build/HTML-Parser-3.55

  CPAN.pm: Going to build G/GA/GAAS/HTML-Parser-3.55.tar.gz

Checking if your kit is complete...
Looks good
Writing Makefile for HTML::Parser
---- Unsatisfied dependencies detected during [G/GA/GAAS/HTML-Parser-3.55.tar.gz] -----
    HTML::Tagset
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* HTML::HeadParser installation failed.
**\* Installing HTML::LinkExtor...
Running install for module HTML::LinkExtor
Running make for G/GA/GAAS/HTML-Parser-3.55.tar.gz
  Is already unwrapped into directory /home/hctest/.cpan/build/HTML-Parser-3.55
  Delayed until after prerequisites
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* HTML::LinkExtor installation failed.
**\* Installing HTTP::Request::Common...
Running install for module HTTP::Request::Common
Running make for G/GA/GAAS/libwww-perl-5.805.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/G/GA/GAAS/libwww-perl-5.805.tar.gz ok
libwww-perl-5.805/
libwww-perl-5.805/lib/
libwww-perl-5.805/lib/HTTP/
libwww-perl-5.805/lib/HTTP/Cookies.pm
libwww-perl-5.805/lib/HTTP/Status.pm
libwww-perl-5.805/lib/HTTP/Cookies/
libwww-perl-5.805/lib/HTTP/Cookies/Netscape.pm
libwww-perl-5.805/lib/HTTP/Cookies/Microsoft.pm
libwww-perl-5.805/lib/HTTP/Request/
libwww-perl-5.805/lib/HTTP/Request/Common.pm
libwww-perl-5.805/lib/HTTP/Headers/
libwww-perl-5.805/lib/HTTP/Headers/Auth.pm
libwww-perl-5.805/lib/HTTP/Headers/ETag.pm
libwww-perl-5.805/lib/HTTP/Headers/Util.pm
libwww-perl-5.805/lib/HTTP/Request.pm
libwww-perl-5.805/lib/HTTP/Response.pm
libwww-perl-5.805/lib/HTTP/Date.pm
libwww-perl-5.805/lib/HTTP/Daemon.pm
libwww-perl-5.805/lib/HTTP/Message.pm
libwww-perl-5.805/lib/HTTP/Negotiate.pm
libwww-perl-5.805/lib/HTTP/Headers.pm
libwww-perl-5.805/lib/LWP/
libwww-perl-5.805/lib/LWP/Protocol/
libwww-perl-5.805/lib/LWP/Protocol/nntp.pm
libwww-perl-5.805/lib/LWP/Protocol/ftp.pm
libwww-perl-5.805/lib/LWP/Protocol/mailto.pm
libwww-perl-5.805/lib/LWP/Protocol/GHTTP.pm
libwww-perl-5.805/lib/LWP/Protocol/nogo.pm
libwww-perl-5.805/lib/LWP/Protocol/file.pm
libwww-perl-5.805/lib/LWP/Protocol/loopback.pm
libwww-perl-5.805/lib/LWP/Protocol/data.pm
libwww-perl-5.805/lib/LWP/Protocol/https10.pm
libwww-perl-5.805/lib/LWP/Protocol/http10.pm
libwww-perl-5.805/lib/LWP/Protocol/https.pm
libwww-perl-5.805/lib/LWP/Protocol/gopher.pm
libwww-perl-5.805/lib/LWP/Protocol/cpan.pm
libwww-perl-5.805/lib/LWP/Protocol/http.pm
libwww-perl-5.805/lib/LWP/RobotUA.pm
libwww-perl-5.805/lib/LWP/Authen/
libwww-perl-5.805/lib/LWP/Authen/Ntlm.pm
libwww-perl-5.805/lib/LWP/Authen/Digest.pm
libwww-perl-5.805/lib/LWP/Authen/Basic.pm
libwww-perl-5.805/lib/LWP/MemberMixin.pm
libwww-perl-5.805/lib/LWP/media.types
libwww-perl-5.805/lib/LWP/DebugFile.pm
libwww-perl-5.805/lib/LWP/Debug.pm
libwww-perl-5.805/lib/LWP/UserAgent.pm
libwww-perl-5.805/lib/LWP/MediaTypes.pm
libwww-perl-5.805/lib/LWP/Protocol.pm
libwww-perl-5.805/lib/LWP/ConnCache.pm
libwww-perl-5.805/lib/LWP/Simple.pm
libwww-perl-5.805/lib/WWW/
libwww-perl-5.805/lib/WWW/RobotRules/
libwww-perl-5.805/lib/WWW/RobotRules/AnyDBM_File.pm
libwww-perl-5.805/lib/WWW/RobotRules.pm
libwww-perl-5.805/lib/File/
libwww-perl-5.805/lib/File/Listing.pm
libwww-perl-5.805/lib/Net/
libwww-perl-5.805/lib/Net/HTTP/
libwww-perl-5.805/lib/Net/HTTP/Methods.pm
libwww-perl-5.805/lib/Net/HTTP/NB.pm
libwww-perl-5.805/lib/Net/HTTP.pm
libwww-perl-5.805/lib/Net/HTTPS.pm
libwww-perl-5.805/lib/Bundle/
libwww-perl-5.805/lib/Bundle/LWP.pm
libwww-perl-5.805/lib/LWP.pm
libwww-perl-5.805/lib/HTML/
libwww-perl-5.805/lib/HTML/Form.pm
libwww-perl-5.805/t/
libwww-perl-5.805/t/base/
libwww-perl-5.805/t/base/headers-util.t
libwww-perl-5.805/t/base/status.t
libwww-perl-5.805/t/base/cookies.t
libwww-perl-5.805/t/base/headers-etag.t
libwww-perl-5.805/t/base/http.t
libwww-perl-5.805/t/base/date.t
libwww-perl-5.805/t/base/listing.t
libwww-perl-5.805/t/base/response.t
libwww-perl-5.805/t/base/headers-auth.t
libwww-perl-5.805/t/base/common-req.t
libwww-perl-5.805/t/base/protocols.t
libwww-perl-5.805/t/base/message.t
libwww-perl-5.805/t/base/mediatypes.t
libwww-perl-5.805/t/base/ua.t
libwww-perl-5.805/t/base/headers.t
libwww-perl-5.805/t/base/negotiate.t
libwww-perl-5.805/t/base/message-old.t
libwww-perl-5.805/t/base/message-parts.t
libwww-perl-5.805/t/local/
libwww-perl-5.805/t/local/get.t
libwww-perl-5.805/t/local/autoload.t
libwww-perl-5.805/t/local/http.t
libwww-perl-5.805/t/local/http-get.t
libwww-perl-5.805/t/local/autoload-get.t
libwww-perl-5.805/t/local/protosub.t
libwww-perl-5.805/t/live/
libwww-perl-5.805/t/live/jigsaw-te.t
libwww-perl-5.805/t/live/validator.t
libwww-perl-5.805/t/live/jigsaw-md5-get.t
libwww-perl-5.805/t/live/jigsaw-neg.t
libwww-perl-5.805/t/live/jigsaw-auth-d.t
libwww-perl-5.805/t/live/apache-listing.t
libwww-perl-5.805/t/live/jigsaw-md5.t
libwww-perl-5.805/t/live/jigsaw-neg-get.t
libwww-perl-5.805/t/live/jigsaw-chunk.t
libwww-perl-5.805/t/live/jigsaw-auth-b.t
libwww-perl-5.805/t/live/activestate.t
libwww-perl-5.805/t/net/
libwww-perl-5.805/t/net/http-get.t
libwww-perl-5.805/t/net/cgi-bin/
libwww-perl-5.805/t/net/cgi-bin/nph-slowdata
libwww-perl-5.805/t/net/cgi-bin/moved
libwww-perl-5.805/t/net/cgi-bin/test
libwww-perl-5.805/t/net/cgi-bin/slowread
libwww-perl-5.805/t/net/cgi-bin/timeout
libwww-perl-5.805/t/net/proxy.t
libwww-perl-5.805/t/net/http-post.t
libwww-perl-5.805/t/net/moved.t
libwww-perl-5.805/t/net/http-timeout.t
libwww-perl-5.805/t/net/config.pl.dist
libwww-perl-5.805/t/net/mirror.t
libwww-perl-5.805/t/robot/
libwww-perl-5.805/t/robot/rules.t
libwww-perl-5.805/t/robot/ua.t
libwww-perl-5.805/t/robot/ua-get.t
libwww-perl-5.805/t/robot/rules-dbm.t
libwww-perl-5.805/t/README
libwww-perl-5.805/t/html/
libwww-perl-5.805/t/html/form-param.t
libwww-perl-5.805/t/html/form.t
libwww-perl-5.805/t/TEST
libwww-perl-5.805/bin/
libwww-perl-5.805/bin/lwp-rget
libwww-perl-5.805/bin/lwp-mirror
libwww-perl-5.805/bin/lwp-download
libwww-perl-5.805/bin/lwp-request
libwww-perl-5.805/Makefile.PL
libwww-perl-5.805/talk-to-ourself
libwww-perl-5.805/lwptut.pod
libwww-perl-5.805/MANIFEST
libwww-perl-5.805/README.SSL
libwww-perl-5.805/Changes
libwww-perl-5.805/AUTHORS
libwww-perl-5.805/README
libwww-perl-5.805/lwpcook.pod
Removing previously used /home/hctest/.cpan/build/libwww-perl-5.805

  CPAN.pm: Going to build G/GA/GAAS/libwww-perl-5.805.tar.gz

This package comes with some sample programs that I can try
to install in /usr/bin.

   Note that you can avoid these questions by passing
   the '-n' option to 'Makefile.PL'.

Do you want to install lwp-request? [y] Do you want to install lwp-mirror? [y] Do you want to install lwp-rget? [y] Do you want to install lwp-download? [y] 
The lwp-request program will use the name it is invoked with to
determine what HTTP method to use.  I can set up alias for the most
common HTTP methods.  These alias are also installed in
/usr/bin.

Do you want to install the GET alias? [n] Do you want to install the HEAD alias? [n] Do you want to install the POST alias? [n] 
Checking for URI........... ok
Checking for HTML::Parser.. failed
Can't locate HTML/HeadParser.pm in @INC (@INC contains: /usr/lib/perl5/5.8/cygwin /usr/lib/perl5/5.8 /usr/lib/perl5/site_perl/5.8/cygwin /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/site_perl/5.8/cygwin /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/vendor_perl/5.8/cygwin /usr/lib/perl5/vendor_perl/5.8 /usr/lib/perl5/vendor_perl/5.8/cygwin /usr/lib/perl5/vendor_perl/5.8 .) at Makefile.PL line 167, <STDIN> line 7.

The HTML::Parser is needed to extract correct base URI information from
HTML so that we can resolve relative links correctly.  The HTML::Form
module also need HTML::TokeParser to work.

Checking for MIME::Base64.. ok
Checking for Net::FTP...... ok
Checking for Digest::MD5 .. ok
The missing modules can be obtained from CPAN.  Visit
URL:http://www.perl.com/CPAN/ to find a CPAN site near you.

Checking if your kit is complete...
Looks good
Writing Makefile for LWP
---- Unsatisfied dependencies detected during [G/GA/GAAS/libwww-perl-5.805.tar.gz] -----
    HTML::Parser
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* HTTP::Request::Common installation failed.
**\* Installing HoneyClient::Util::Config...
**\* Could not find a version 0 or above for HoneyClient::Util::Config; skipping.
**\* Installing HoneyClient::Util::SOAP...
**\* Could not find a version 0 or above for HoneyClient::Util::SOAP; skipping.
**\* Installing LWP::UserAgent...
Running install for module LWP::UserAgent
Running make for G/GA/GAAS/libwww-perl-5.805.tar.gz
  Is already unwrapped into directory /home/hctest/.cpan/build/libwww-perl-5.805
  Delayed until after prerequisites
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* LWP::UserAgent installation failed.
**\* Installing Term::ProgressBar...
Running install for module Term::ProgressBar
Running make for F/FL/FLUFFY/Term-ProgressBar-2.09.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/F/FL/FLUFFY/Term-ProgressBar-2.09.tar.gz ok
Term-ProgressBar-2.09/
Term-ProgressBar-2.09/BUGS
Term-ProgressBar-2.09/Build.PL
Term-ProgressBar-2.09/Changes
Term-ProgressBar-2.09/configure
Term-ProgressBar-2.09/examples/
Term-ProgressBar-2.09/examples/powers
Term-ProgressBar-2.09/examples/powers2
Term-ProgressBar-2.09/examples/powers3
Term-ProgressBar-2.09/examples/powers4
Term-ProgressBar-2.09/examples/powers5
Term-ProgressBar-2.09/INSTALL
Term-ProgressBar-2.09/lib/
Term-ProgressBar-2.09/lib/Term/
Term-ProgressBar-2.09/lib/Term/ProgressBar.pm
Term-ProgressBar-2.09/make-pm
Term-ProgressBar-2.09/Makefile.PL
Term-ProgressBar-2.09/MANIFEST
Term-ProgressBar-2.09/MANIFEST.SKIP
Term-ProgressBar-2.09/META.yml
Term-ProgressBar-2.09/README
Term-ProgressBar-2.09/SIGNATURE
Term-ProgressBar-2.09/t/
Term-ProgressBar-2.09/t/0-signature.t
Term-ProgressBar-2.09/t/compat.t
Term-ProgressBar-2.09/t/eta-linear.t
Term-ProgressBar-2.09/t/name.t
Term-ProgressBar-2.09/t/test.pm
Term-ProgressBar-2.09/t/v1-message.t
Term-ProgressBar-2.09/t/v2-message.t
Term-ProgressBar-2.09/t/v2-mobile.t
Term-ProgressBar-2.09/t/v2-simple.t
Term-ProgressBar-2.09/t/zero.t
Removing previously used /home/hctest/.cpan/build/Term-ProgressBar-2.09

  CPAN.pm: Going to build F/FL/FLUFFY/Term-ProgressBar-2.09.tar.gz

Checking if your kit is complete...
Looks good
Writing Makefile for Term-ProgressBar
---- Unsatisfied dependencies detected during [F/FL/FLUFFY/Term-ProgressBar-2.09.tar.gz] -----
    Class::MethodMaker
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* Term::ProgressBar installation failed.
**\* Installing Test::Exception...
Running install for module Test::Exception
Running make for A/AD/ADIE/Test-Exception-0.24.tar.gz
Checksum for /home/hctest/.cpan/sources/authors/id/A/AD/ADIE/Test-Exception-0.24.tar.gz ok
Test-Exception-0.24/
Test-Exception-0.24/Build.PL
Test-Exception-0.24/Changes
Test-Exception-0.24/lib/
Test-Exception-0.24/lib/Test/
Test-Exception-0.24/lib/Test/Exception.pm
Test-Exception-0.24/Makefile.PL
Test-Exception-0.24/MANIFEST
Test-Exception-0.24/META.yml
Test-Exception-0.24/README
Test-Exception-0.24/t/
Test-Exception-0.24/t/caller.t
Test-Exception-0.24/t/developer/
Test-Exception-0.24/t/developer/documented.t
Test-Exception-0.24/t/developer/perlcritic.t
Test-Exception-0.24/t/developer/perlcriticrc
Test-Exception-0.24/t/developer/pod.t
Test-Exception-0.24/t/developer/spelling.t
Test-Exception-0.24/t/Exception.t
Test-Exception-0.24/t/import.t
Test-Exception-0.24/t/isa.t
Test-Exception-0.24/t/lives_and.t
Test-Exception-0.24/t/preserve.t
Test-Exception-0.24/t/return.t
Test-Exception-0.24/t/rt.t
Test-Exception-0.24/t/stacktrace.t
Test-Exception-0.24/t/throws_ok.t
Removing previously used /home/hctest/.cpan/build/Test-Exception-0.24

  CPAN.pm: Going to build A/AD/ADIE/Test-Exception-0.24.tar.gz

Checking if your kit is complete...
Looks good
Writing Makefile for Test::Exception
---- Unsatisfied dependencies detected during [A/AD/ADIE/Test-Exception-0.24.tar.gz] -----
    Test::Builder
    Sub::Uplevel
    Test::Builder::Tester
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* Test::Exception installation failed.
**\* Module::AutoInstall installation finished.
Writing /usr/lib/perl5/site_perl/5.8/cygwin/auto/HoneyClient/Agent/.packlist
Appending installation info to /usr/lib/perl5/5.8/cygwin/perllocal.pod
}}}

**Comment by dkindlund** [comment]
Author: kindlund

**Comment by dkindlund** [comment]
Author: kindlund
Updated HoneyClient::Util dependency list.  Leaving ticket open until all dependencies are found.

**Comment by dkindlund** [comment]
Author: xkovah
The first instance of a prereq install failing is this:
{{{
Writing Makefile for Data::Compare
---- Unsatisfied dependencies detected during [D/DC/DCANTRELL/Data-Compare-0.14.tar.gz] -----
    File::Find::Rule
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
**\* Data::Compare installation failed.
}}}
We can see that it fails to auto-install the File::Find::Rule dependency for Data-Compare. I have seen this sort of error on occasion due to there not being enough mirrors selected because even sometimes a mirror may not have all the packages or some might be slow while others are fast. It might be helpful if we got the person's configuration by having them do "o conf" from the cpan prompt and then sending us the output.

**Comment by dkindlund** [comment]
Author: xkovah
btw, are we sure we want to list the prerequisites of the prerequisites (i.e. File::Find::Rule) on the page? Because when you install Data::Compare from CPAN it seems to correctly install File::Find::Rule, it's just from the make which it doesn't seem to be working. Therefore theoretically the list as given by the "perl Makefile.PL" is the minimum that they would need to manually install.

**Comment by dkindlund** [comment]
Author: kindlund
Replying to [comment:4 xkovah]:

> btw, are we sure we want to list the prerequisites of the prerequisites (i.e. File::Find::Rule) on the page? Because when you install Data::Compare from CPAN it seems to correctly install File::Find::Rule, it's just from the make which it doesn't seem to be working. Therefore theoretically the list as given by the "perl Makefile.PL" is the minimum that they would need to manually install.

Once we get manual installation to work by calling "perl Makefile.PL" directly, then we'll trim the dependency list accordingly.  We need to figure out how to have the manual install process fail preemptively, if '''any''' dependency fails before it.  Currently, when any dependency installation fails, perl happily skips the dependency and tries to install other parent dependencies -- this is not an ideal situation.

**Comment by dkindlund** [comment]
Author: xkovah
{{{
I'm going to put this in here since I know it sends the email. This is mainly going to deal with my install but it has parallels for the submitted failed install above.

The first time I tried to do my own install I did it from the first user I created rather than the Administrator account. This is because the UserGuide didn't used to mention the Administrator account until step 10, which would have been after where you get the above error (and even then it's more implied than stated, and one could interpret it as meaning that the make install was just supposed to put stuff in the admin's account). Therefore I can also see that in the above error, the user was using a non-administrator account by the name of hctest. So I added the explicit instruction to use the Administrator in step 4. Also, when I had first seen the step which was suggesting that I should probably be using the Administrator account, I switched over and then had things like XML::Tidy which would silently fail even though they said they were OK from the cpan install prompt. This turned out to be a permissions problem in that my cygwin perl was owned by my initial user account and interestingly the Administrator account couldn't write to it (cause of the way the "unix" permissions were set). Therefore, I eventually found that I needed to do an rm -rf /usr/lib/perl5 and reinstall perl from the cygwin setup to ensure that it's permissions were OK for the Administrator account. If the person who had the above problem wants to start over, logging in to the Administrator account and doing that rm would probably be the fastest course of action.

Also at some point I had tracked down that some prereq for some package was probably failing due to the lack of ExtUtils::PkgConfig. I remembered that that was probably necessary to be installed through cygwin, but it wasn't in the instructions, so I have subsequently added the extras to UserGuide

Also, one note which I will be following up on later is that after perl Makefile.PL, "make" seems to rely on the Module::AutoInstall, that might be what is failing to properly install the prereqs

The below is just a log I made while I was trying a new configuration to get the Agent installed on the XP VM.

---

I installed perl, the perl ExtUtils-*, and the perl win32, and also wget, gcc, make

also I manually configured cpan
also I did "install Bundle::CPAN"  when asked about YAML I said yes, and when asked about Compress-Raw-Zlib I said no (which made some cascading failures)
and "reload CPAN" and then I quit and then I went back in and configured Build.PL with all the default and then did the necessary "o conf commit"

then I tried "perl Makefile.PL" for Honeyclient:Util first
it was missing Log::Log4perl, SOAP::Lite, SOAP::Transport::HTTP, XML::Tidy, and XML::XPath (5 things)

I did a make and SOAP::Lite bombed out as usual, so I installed LWP::UserAgent via CPAN as usual

After that I tried to run "perl Makefile.PL" again. This time the only things which weren't installed were SOAP::Lite and XML::Tidy (2 things, so 3 installed ok)

installed SOAP::Lite successfully through cpan

now only XML::Tidy was showing from "perl Makefile.PL"

successfully installed XML::Tidy through CPAN

now moved on to HoneyClient::Agent dir

ran "perl Makefile.PL" and it was missing Data::Compare, Data::Diff, Data::Structure::Util, Filesys::CygwinPaths, Parse::Yapp::Driver, Search::Binary, Term::ProgressBar, Test::Exception (9 things)

ran make

After that I tried to run "perl Makefile.PL" again. This time the only things which weren't installed were  Data::Compare, Data::Structure::Util, Term::ProgressBar, Test::Exception (5 things, so 4 installed ok)

successfully installed Data::Compare from cpan

successfully installed Data::Structure::Util from cpan

DateTime::HiRes fails, seemingly due to DateTime::Locale
DateTime::Locale refuses to install through cpan

download DateTime::Locale and install manually
DateTime::HiRes now installs through cpan

successfully installed Term::ProgressBar from cpan

successfully installed Test::Exception from cpan

HoneyClient::Agent now installs fine via make and make install

---

SUMMARY of the only required manual installs (everything else installed fine via make)
(for Utils on windows)
LWP::UserAgent from cpan before installing SOAP::Lite from cpan
XML::Tidy from cpan

(for Agent)
Data::Compare from cpan
Data::Structure::Util from cpan
DateTime::Locale manually before installing DateTime::HiRes from cpan
Term::ProgressBar from cpan
Test::Exception from cpan

(for Utils on SuSE 10.2)
only XML::Tidy had to be manually installed via cpan

(for Manager on my SuSE 10.2 system)
DateTime::Locale again had to be downloaded and installed to get DateTime::HiRes, but everything else was already there.

So I've gotten through the Agent install, but now I'm stymied on the correct way to set up the vmware server, therefore I think ticket #37 is now a higher priority.
}}}
